File: frontend/src/types/category.ts
/**
 * TypeScript types for Category entities
 */

export interface Category {
  id: string
  name: string
  slug: string
  created_by: string
  created_at: string
  video_count: number
}

export interface CategoryCreate {
  name: string
}

export interface CategoryUpdate {
  name: string
}

export interface CategoryListResponse {
  categories: Category[]
  total: number
}

File: backend/app/models/category.py
from sqlalchemy import Column, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
import uuid
from app.database import Base


class Category(Base):
    __tablename__ = "categories"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(100), unique=True, nullable=False, index=True)
    slug = Column(String(100), unique=True, nullable=False, index=True)
    created_by = Column(
        String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False
    )
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)

    # Relationships
    creator = relationship("User", back_populates="categories")
    videos = relationship(
        "Video", back_populates="category", cascade="all, delete-orphan"
    )

    def __repr__(self):
        return f"<Category(id={self.id}, name={self.name}, slug={self.slug})>"
